pkg/monitor/cluster/debugpods.go

```diff
@@ -0,0 +1,69 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+
+	"github.com/Azure/ARO-RP/pkg/util/stringutils"
+)
+
+// Each cluster is being checked every 1 minute. emitDebugPodsCount
+// lists all events (age < 70 seconds) in the default namespace
+// in order to detect recent events of debug-pod creation.
+func (mon *Monitor) emitDebugPodsCount(ctx context.Context) error {
+	events, err := mon.cli.CoreV1().Events("default").List(ctx, metav1.ListOptions{
+		FieldSelector: fields.SelectorFromSet(map[string]string{
+			"reason":              "Started",
+			"involvedObject.kind": "Pod",
+			// oc debug node creates a pod with a "container-00"
+			// container name.
+			"involvedObject.fieldPath": "spec.containers{container-00}",
+		}).String(),
+	})
+	if err != nil {
+		return err
+	}
+
+	nodes, err := mon.cli.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	podNames := getDebugPodNames(nodes.Items)
+
+	count := countDebugPods(podNames, events.Items)
+	mon.emitGauge("debugpods.count", int64(count), nil)
+
+	return nil
+}
+
+func countDebugPods(debugPods []string, events []corev1.Event) int {
+	count := 0
+	for _, e := range events {
+		if eventIsNew(e) && stringutils.Contains(debugPods, e.InvolvedObject.Name) {
+			count++
+		}
+	}
+	return count
+}
```

Comment:
I really like the split in narrow functions. 😄 

If we know that corev1.Event.LastTimeStamp is deprecated in newer k8s releases and we will need to update this check when ARO's k8s version is upgraded, I think we can modify the code a bit to be easier to maintain.

What about something like:
```go
func countDebugPods(debugPods []string, events []corev1.Event) int {
	count := 0
	for _, e := range events {
		// corev1.Event.LastTimeStamp is deprecated in newer k8s releases.
		// Update this check when ARO's k8s version is upgraded
		t := e.LastTimestamp.Time
		if eventIsNew(t) && stringutils.Contains(debugPods, e.InvolvedObject.Name) {
			count++
		}
	}
	return count
}
```

Reply:
Note: the reason I put e.LastTimestamp.Time in a different line instead of inlining it in the function call below it is to have a narrower scope of change in the future and not affect lines that shouldn't be affected by that type of change.

Reply:
is it being deprecated? do you have a link to it?

Reply:
I just trusted the comment on [pkg/monitor/cluster/debugpods.go](https://github.com/Azure/ARO-RP/pull/2656/files#diff-e6aaf68ce82f0311e70154300b03c0a6a7f8c3cb614795f64289a782bf9db4edR58) 😄

Reply:
If we refactor it like this, the function name `eventIsNew` doesn't make sense anymore. I would say it is actually better to only change the lower level function `eventIsNew` if `LastTimestamp` is removed.

Reply:
@AldoFusterTurpin the deprecation issue I referred to was closed several years ago, so these fields will not be deprecated. Sorry for this :)